block/fork.go

```diff
@@ -108,3 +115,62 @@ func (m *Manager) freezeNode(ctx context.Context) {
 	err := fmt.Errorf("node frozen due to fork update")
 	uevent.MustPublish(ctx, m.Pubsub, &events.DataHealthStatus{Error: err}, events.HealthStatusList)
 }
+
+// forkNeeded returns true if the fork file exists
+func (m *Manager) forkNeeded() bool {
+	if _, err := types.LoadInstructionFromDisk(m.RootDir); err == nil {
+		return true
+	}
+
+	return false
+}
+
+// handleSequencerForkTransition handles the sequencer fork transition
+func (m *Manager) handleSequencerForkTransition(instruction types.Instruction) {
+	lastBlock, err := m.Store.LoadBlock(m.State.Height())
+	if err != nil {
+		panic(fmt.Sprintf("load block: height: %d: %v", m.State.Height(), err))
+	}
+
+	var consensusMsgs []proto.Message
+	if instruction.FaultyDRS != nil {
+		if lastBlock.Header.Version.App == instruction.Revision {
+			panic(fmt.Sprintf("running faulty DRS version %d", *instruction.FaultyDRS))
+		}
+
+		msgUpgradeDRS := &sequencers.MsgUpgradeDRS{
+			DrsVersion: *instruction.FaultyDRS,
+		}
+
+		consensusMsgs = append(consensusMsgs, msgUpgradeDRS)
+	}
+
+	// Always bump the account sequences
+	msgBumpSequences := &sequencers.MsgBumpAccountSequences{Authority: "gov"}
+
+	consensusMsgs = append(consensusMsgs, msgBumpSequences)
+
+	// Create a new block with the consensus messages
+	m.Executor.AddConsensusMsgs(consensusMsgs...)
+
+	block, commit, err := m.produceBlock(true, nil, true)
+	if err != nil {
+		panic(fmt.Sprintf("produce block: %v", err))
+	}
+
+	err = m.applyBlock(block, commit, types.BlockMetaData{Source: types.Produced})
+
+	// Create another block emtpy
+	block, commit, err = m.produceBlock(true, nil, true)
+	if err != nil {
+		panic(fmt.Sprintf("produce empty block: %v", err))
+	}
+
+	err = m.applyBlock(block, commit, types.BlockMetaData{Source: types.Produced})
```

Comment:
is there any reason to not gossip these two blocks to the p2p?

Reply:
if there is no specific reason to not gossiping the blocks, i would call the already existing ProduceApplyGossipBlock() func.